server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/node/FixedLengthStarTreeNode.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.index.compositeindex.datacube.startree.node;
+
+import org.apache.lucene.store.RandomAccessInput;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+import static org.opensearch.index.compositeindex.datacube.startree.utils.StarTreeUtils.ALL;
+
+/**
+ * Fixed Length implementation of {@link StarTreeNode}
+ *
+ * @opensearch.experimental
+ */
+public class FixedLengthStarTreeNode implements StarTreeNode {
+    public static final int NUM_INT_SERIALIZABLE_FIELDS = 6;
```

Comment:
nit : Lets add some javaDoc here explaining significance of these. In fact the entire class can be revisited when it comes to add javadocs